packages/client/src/createTRPCClient.ts

```diff
@@ -31,32 +31,49 @@ const retryDelay = (attemptIndex: number) =>
   attemptIndex === 0 ? 0 : Math.min(1000 * 2 ** attemptIndex, 30000);
 
 export class TRPCClientError<TRouter extends AnyRouter> extends Error {
-  public readonly json?: Maybe<HTTPErrorResponseEnvelope<TRouter>>;
-  /**
-   * @deprecated will always be `undefined` and will be removed in next major
-   */
+  public readonly result?: Maybe<TRPCProcedureErrorEnvelope<TRouter>>;
   public readonly res?: Maybe<Response>;
   public readonly originalError?: Maybe<Error>;
-  public readonly shape?: HTTPErrorResponseEnvelope<TRouter>['error'];
+  public readonly shape?: TRPCProcedureErrorEnvelope<TRouter>['error'];
 
   constructor(
     message: string,
     {
-      json,
+      result: result,
       originalError,
     }: {
-      json?: Maybe<HTTPErrorResponseEnvelope<TRouter>>;
-      originalError?: Maybe<Error>;
+      result: Maybe<TRPCProcedureErrorEnvelope<TRouter>>;
+      originalError: Maybe<Error>;
     },
   ) {
     super(message);
     this.message = message;
-    this.json = json;
+    this.result = result;
     this.originalError = originalError;
-    this.shape = this.json?.error;
+    this.shape = this.result?.error;
 
     Object.setPrototypeOf(this, TRPCClientError.prototype);
   }
+
+  public static from<TRouter extends AnyRouter>(
+    result: Error | TRPCProcedureErrorEnvelope<TRouter>,
+  ): TRPCClientError<TRouter> {
+    if (!(result instanceof Error)) {
+      return new TRPCClientError<TRouter>((result.error as any).message ?? '', {
```

Comment:
if result isn't a instance of error, how do you know result.error is? I know the types above enforce it, but someone could accidentatlly call this with something casted to any? 

I'm not sure where this is called though. I see its not exported

Reply:
Nevermind I see now where this get called from and I didnt realise this is a method on the error class, I thought it was on the client class. Makes sense